joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/xml/SecureXmlParsing.scala

```diff
@@ -0,0 +1,27 @@
+package io.joern.x2cpg.utils.xml
+
+import javax.xml.parsers.SAXParserFactory
+import scala.xml.{Elem, XML}
+
+object SecureXmlParsing {
+  def parseXml(content: String): Option[Elem] = {
+    try {
+      val spf = SAXParserFactory.newInstance()
+
+      spf.setValidating(false)
+      spf.setNamespaceAware(false)
+      spf.setXIncludeAware(false)
+      spf.setFeature("http://xml.org/sax/features/validation", false)
+      spf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false)
+      spf.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false)
+      spf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false)
+      spf.setFeature("http://xml.org/sax/features/external-parameter-entities", false)
+      spf.setFeature("http://xml.org/sax/features/external-general-entities", false)
+
+      Some(XML.withSAXParser(spf.newSAXParser()).loadString(content))
+    } catch {
+      case _: Throwable =>
+        None
+    }
+  }
+}
```

Comment:
```suggestion
import scala.util.Try
import scala.xml.{Elem, XML}

object SecureXmlParsing {
  def parseXml(content: String): Option[Elem] = {
    Try {
      val spf = SAXParserFactory.newInstance()

      spf.setValidating(false)
      spf.setNamespaceAware(false)
      spf.setXIncludeAware(false)
      spf.setFeature("http://xml.org/sax/features/validation", false)
      spf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false)
      spf.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false)
      spf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false)
      spf.setFeature("http://xml.org/sax/features/external-parameter-entities", false)
      spf.setFeature("http://xml.org/sax/features/external-general-entities", false)

      Some(XML.withSAXParser(spf.newSAXParser()).loadString(content))
    }.toOption
  }
}
```

Reply:
more readable, and ensure's you're not catching stuff you really don't want to catch, like OutOfMemoryError etc. 

I haven't asked scalac for feedback on this yet, but that's the gist.

Reply:
nicer, updated, thx